testgrid/cmd/transfigure/transfigure.sh

```diff
@@ -69,7 +69,7 @@ main() {
     --output "${testgrid_dir}/${testgrid_subdir}/gen-config.yaml"
 
 
-  if ! git diff --quiet ; then
+  if [[ $(git diff --exit-code) -eq 0 ]]; then
```

Comment:
This works fine, except for in the case where the config directory does not already exist. If there are untracked files in the repository, `git diff --exit-code` still returns 0.

This is probably sufficient to fix the current issues, but thought it worth mentioning, especially as there is handling for creating the subdirectory above too:
```
  if [[ ! -d "${testgrid_dir}/${testgrid_subdir}" ]]; then
    echo "Subdirectory ${testgrid_subdir} doesn't exist; creating it" >&2
    mkdir -p "${testgrid_dir}/${testgrid_subdir}"
  fi
```

Reply:
I think the previous (working) iteration of the script also had the same issue, so I'll add my lgtm + a hold - feel free to unhold if you want to fix it later 😄 

/lgtm
/hold